prow/plugins/dog/dog_test.go

```diff
@@ -151,6 +171,23 @@ func TestHttpResponse(t *testing.T) {
 		}
 	}))
 	defer ts.Close()
+
+	// fake server for images
+	ts2 := httptest.NewUnstartedServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if s, ok := contentLength[r.URL.Path]; ok {
+			body := "binary image"
+			w.Header().Set("Content-Length", s)
+			io.WriteString(w, body)
+		} else {
+			t.Errorf("Cannot find content length for %s", r.URL.Path)
+		}
+	}))
+	ts2.Listener.Close()
+	ts2.Listener = listener
```

Comment:
I don't think there is a need to create a separate listener to replace the one provided by the `httptest` server.
It looks like you are doing this so that you can determine the port that the listener is listening to, but you can instead just use the `httptest` server's `URL` field.

Reply:
Well, there is a chicken/egg problem here... as I need the port when declaring the handler function, and ts2.URL is only available after I have already created/started the server by specifying the handler.

Reply:
The port is used by the handler func for `ts` not `ts2`. `ts2`'s handler only captures `contentLength` which doesn't depend on the port. Couldn't `ts2` just be defined before everything else and it's URL used directly instead of replacing the listener?

Reply:
Oh... yes! This is a remain from when I was planning to use only one server.
You are 100% right, I will correct that and use the same strategy for cats if that's ok for you?

Reply:
Yes, that would be great!